src/sentry/static/sentry/app/actionCreators/globalSelection.tsx

```diff
@@ -1,12 +1,69 @@
+import * as ReactRouter from 'react-router';
+import * as Sentry from '@sentry/browser';
 import isInteger from 'lodash/isInteger';
 import omit from 'lodash/omit';
 import qs from 'query-string';
-import * as Sentry from '@sentry/browser';
 
+import {Environment} from 'app/types';
 import {defined} from 'app/utils';
 import {getUtcDateString} from 'app/utils/dates';
 import GlobalSelectionActions from 'app/actions/globalSelectionActions';
 
+/**
+ * Note this is the internal project.id, NOT the slug, but it is the stringified version of it
+ */
+type ProjectId = string | number;
+type EnvironmentId = Environment['id'];
+
+type Options = {
+  /**
+   * List of parameters to remove when changing URL params
+   */
+  resetParams?: string[];
+};
+
+/**
+ * Can be relative time string or absolute (using start and end dates)
+ */
+type DateTimeObject = {
+  start?: Date | string | null;
+  end?: Date | string | null;
+  statsPeriod?: string | null;
+  utc?: string | boolean | null;
+  /**
+   * @deprecated
+   */
+  period?: string | null;
+};
+
+// Get Params type from `getParams` helper
+// type Params = Parameters<typeof getParams>[0];
+
+/**
+ * Cast project ids to strings, as everything is assumed to be a string in URL params
+ *
+ * Discover v1 uses a different interface, and passes slightly different datatypes e.g. Date for dates
+ */
+// type UrlParams = {project?: string[]; environment?: EnvironmentId[]} & Params;
+type UrlParams = {
+  project?: ProjectId[];
+  environment?: EnvironmentId[];
+} & DateTimeObject & {
+    // TODO(discoverv1): This can be back to `ParamValue` when we remove Discover
+    [others: string]: any;
+  };
+
+/**
+ * TODO(ts): I think `InjectedRouter` is typed incorrectly, if you inspect in the application,
+ * you'll see that `router` also includes `WithRouterProps`
```

Comment:
Could this be because of the old typings version?

Reply:
Nope, we are up to date